doc/design/diag_attribute_error.md

```diff
@@ -0,0 +1,96 @@
+# Diagnostic Attribute Error
+
+## Motivation
+
+SQLFlow extended SQL syntax to describe an end-to-end machine learning job, for a typical SQL program:
+
+``` python
+SELECT * FROM iris.train
+TO TRAIN DNNClassifier
+WITH
+  model.n_classes = 2,
+  model.hidden_units = [128, 64]
+LABEL class
+INTO my_model;
+
+SELECT * FROM iris.test
+TO PREDICT iris.pred.class
+USING my_model;
+```
+
+SQLFlow compiles the above SQL program into an execution plan and runs it.  As the TRAIN statement above, SQLFlow uses `TO TRAIN CLAUSE`  to train a specific model called  `DNNClassifier`, using `WITH CLAUSE` to configure the training arguments. 
+
+Sometimes users may make some configuration mistake on `WITH CLAUSE`, then the job would fault during execution and return some uncertain error message.
+
+This documentation issued a way that adding extra contract in [docstring](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html) to contract the arguments, and this can achieve three advantage at least:
+
+1. Early testing, we can do early testing before running the job; users can wait less time and cluster save resources.
+2. More accurate diagnostic message.
+3. Using docstring is Keras or Tensorflow native; users don't need to modify the model code.
+
+## Design
+
+The docstring of a function or a class can include argument documentation and contracts.
```

Comment:
Do you want to say "We want to document the compiler-readable description of model parameters in the docstring of Python function or class that define a model"?